neuro_config_client/client.py

```diff
@@ -276,6 +282,207 @@ async def remove_storage(
+            return np_templates
+
+    async def add_node_pool(
+        self,
+        cluster_name: str,
+        node_pool: NodePool,
+        *,
+        token: str | None = None,
+        start_deployment: bool = True,
+    ) -> Cluster:
+        """Add new node pool to the existing cluster.
+        Cloud provider should be already set up.
+
+        Make sure you use one of the available node pool templates by providing its ID,
+            if the cluster is deployed in public cloud (AWS / GCP / Azure / VCD).
+
+        Args:
+            cluster_name (str): Name of the cluster within the platform.
+            node_pool (NodePool): Node pool instance.
+                For templates, you could use template.to_node_pool() method
+            token (str | None, optional): User token to perform changes.
+                Should has write access to cluster://{cluster_name}/cloud_provider role.
+            start_deployment (bool, optional): Start applying changes. Defaults to True.
+
+        Returns:
+            Cluster: Cluster instance with applied changes
+        """
+        assert self._client
+
+        cluster = await self.get_cluster(cluster_name, token=token)
+        assert cluster.cloud_provider
+
+        url = self._clusters_url / cluster.name / "cloud_provider/node_pools"
+        headers = self._create_headers(token=token)
+        payload = self._payload_factory.create_node_pool(
+            node_pool, cluster.cloud_provider.type
+        )
+        async with self._client.post(
+            url.with_query(start_deployment=str(start_deployment).lower()),
+            headers=headers,
+            json=payload,
+        ) as response:
+            response.raise_for_status()
+            resp_payload = await response.json()
+            return self._entity_factory.create_cluster(resp_payload)
+
+    async def put_node_pool(
+        self,
+        cluster_name: str,
+        node_pool_name: str,
+        *,
+        role: NodeRole | None = None,
+        min_size: int | None = None,
+        max_size: int | None = None,
+        idle_size: int | None = None,
+        machine_type: str | None = None,
+        cpu: float | None = None,
+        available_cpu: float | None = None,
+        memory_mb: int | None = None,
+        available_memorty_mb: int | None = None,
+        disk_size_gb: int | None = None,
+        disk_type: str | None = None,
+        gpu: int | None = None,
+        gpu_model: str | None = None,
+        price: Decimal | None = None,
+        currency: str | None = None,
+        is_preemptible: bool | None = None,
+        zones: tuple[str] | None = None,
+        token: str | None = None,
+        start_deployment: bool = True,
+    ) -> Cluster:
+        assert self._client
+
+        current_np = await self.get_node_pool(cluster_name, node_pool_name, token=token)
+        cluster = await self.get_cluster(cluster_name, token=token)
+        assert cluster.cloud_provider
+
+        new_np = NodePool(
```

Comment:
This won't work, server validation will fail. If we need to patch node pools then let's add PATCH endpoint.

Reply:
It worked when I cut off unneeded info at the payload factory, but you are right.
Added feature request, https://github.com/neuro-inc/platform-config/issues/1276, updated method.